### PR TITLE
Don't treat emails as relative URLs in the README

### LINKED ISSUE #454: Emails are treated as relative URLs in readme
https://github.com/kelu124/bomanz

|v1|v2|
|---|---|
|image|image|

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
Fixes #454

### CODE REVIEW

File: processor/src/utils.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export function findKicadSchematic(inputDir, files, kitspaceYaml) {
 }
 
 export function isRelativeUrl(uri: string) {
-  return !uri.match(/^https?:\/\//)
+  return !uri.match(/^https?:\/\/|^mailto:/)
```
Comment (kasbah) [inline processor/src/utils.ts:62]: I think it makes sense to make this generic over the protocol. We might get the same problem with e.g. `ssh://` links or something else.

Comment (AbdulrhmnGhanem) [reply processor/src/utils.ts]: I found that the sanitizer only allows `http(s)` and `mailto`. I changed that :point_down:
https://github.com/kitspace/kitspace-v2/blob/f51a0792074fdbb19caaecccb1e5a68ebfb4907c/processor/src/tasks/processReadme/renderMarkdown.ts#L30

And changed the `isRelativeUrl` accordingly.

Do we need to allow any other protocols?

Comment (kasbah) [reply processor/src/utils.ts]: No, we don't need them. It's fine and in fact better if the others don't get "linkified" at all.

Comment (kasbah) [reply processor/src/utils.ts]: Actually, wait, maybe it handy to all of these linkified? I don't think they are used very much but might as well?

Comment (AbdulrhmnGhanem) [reply processor/src/utils.ts]: They are already being linkified, no extra work is required. The only thing I did is stop treating them as relative URLs.

Comment (kasbah) [reply processor/src/utils.ts]: But before the sanitizer would strip them out completely?

File: processor/test/unit/readme_processing.test.ts [hunk 2]
```diff
@@ -20,4 +19,34 @@ describe('readme processing', function () {
...
+      originalUrl: 'https://github.com/kelu124/bomanz/',
+      defaultBranch: 'main',
+    }
+    const html = await renderMarkdown(args)
+    expect(html).toContain('href="mailto:kelu124@gmail.com"')
+  })
+
+  it('does not treat urls starting with a TCP protocol as a relative URL', async function () {
+    const readmeWitSshURL =
+      "I don't know why someone would put ssh url in a readme: [but just in case](ssh://git@github.com:kitspace/kitspace-v2.git)"
```
Comment (kasbah) [inline processor/test/unit/readme_processing.test.ts:40]: ^_^

File: processor/src/utils.ts (hunk 1 above)
Comment (AbdulrhmnGhanem) [reply processor/src/utils.ts]: Yes!

File: processor/test/unit/readme_processing.test.ts [hunk 3]
```diff
@@ -20,4 +19,34 @@ describe('readme processing', function () {
     // if it throws an error this test will fail
     await renderMarkdown(args)
   })
+
+  it('does not treat mailto as a relative URL', async function () {
+    const readmeWithMailto = 'Copyright Kelu124 (kelu124@gmail.com ) 2015-2018'
```
Comment (kasbah) [inline processor/test/unit/readme_processing.test.ts:24]:
```suggestion
    const readmeWithMailto = 'Copyright Example (example124@example.com ) 2015-2018'
```

File: processor/test/unit/readme_processing.test.ts [hunk 4]
```diff
@@ -20,4 +19,34 @@ describe('readme processing', function () {
...
+    const args = {
+      rawMarkdown: readmeWithMailto,
+      ownerName: 'test',
+      repoName: 'test',
+      readmeFolder: 'test',
+      originalUrl: 'https://github.com/kelu124/bomanz/',
+      defaultBranch: 'main',
+    }
+    const html = await renderMarkdown(args)
+    expect(html).toContain('href="mailto:kelu124@gmail.com"')
```
Comment (kasbah) [inline processor/test/unit/readme_processing.test.ts:35]:
```suggestion
    expect(html).toContain('href="mailto:example124@example.com"')
```

**Pull request merged**